mocks/sender.go

```diff
@@ -0,0 +1,88 @@
+package mocks
+
+import (
+	"context"
+	"sync"
+
+	pb "github.com/topfreegames/protos/eventsgateway/grpc/generated"
+)
+
+type MockSender struct {
+	totalSent           int
+	totalCalls          int
+	failureIndexesOrder [][]int64
+	firstCallEvents     []*pb.Event
+	eventsIdsFreqs      map[string]int
+	mutex               *sync.Mutex
+}
+
+func NewMockSender() *MockSender {
+	return &MockSender{
+		failureIndexesOrder: [][]int64{},
+		eventsIdsFreqs:      map[string]int{},
+		mutex:               &sync.Mutex{},
+	}
+}
+
+func (ms *MockSender) SendEvents(
+	ctx context.Context,
+	events []*pb.Event,
+) []int64 {
+	ms.mutex.Lock()
+	defer ms.mutex.Unlock()
+	defer func() { ms.totalCalls++ }()
+	ms.totalSent += len(events)
+	if ms.totalCalls == 0 {
+		ms.firstCallEvents = events
+	}
+	for _, e := range events {
+		ms.eventsIdsFreqs[e.Id]++
+	}
+	if len(ms.failureIndexesOrder) > ms.totalCalls {
+		ret := ms.failureIndexesOrder[ms.totalCalls]
+		ms.totalSent -= len(ret)
+		return ret
+	}
+	return []int64{}
```

Comment:
nit: return nil? it's shorter

Reply:
not sure that saving a few chars is better than safe guarding against nil pointer exceptions (not that it would happen in this case)

Reply:
nil and []int64{} are synonyms in this case, it's perfectly safe.